### ISSUE TITLE
[Feature Request]:  Adding a function to determine whether the metadata is encrypted

### ISSUE THREAD

**Issue by ygqs250** [rikyoz/bit7z#164]
### Feature description

Adding a function to determine whether the metadata is encrypted? Now bit7z can only use exceptions to determine whether the metadata is encrypted.

### Code of Conduct

- [X] By submitting this issue, I agree to follow bit7z's [Code of Conduct](https://github.com/rikyoz/bit7z/blob/master/CODE_OF_CONDUCT.md)

**Comment by rikyoz** [comment]
Hi!

Thanks for the suggestion!

I've just pushed a commit (https://github.com/rikyoz/bit7z/commit/d19423420df7ab6da8bbf2b83076a25b224bafb9) that adds two static functions to the `BitArchiveReader` class:

```cpp
template< typename T >
auto BitArchiveReader::isHeaderEncrypted( const Bit7zLibrary& lib,
                                          T&& in_archive,
                                          const BitInFormat& format ) -> bool;

template< typename T >
auto BitArchiveReader::isEncrypted( const Bit7zLibrary& lib,
                                    T&& in_archive,
                                    const BitInFormat& format ) -> bool;
```

They're template functions so that they can accept either a file path (string), a buffer, or a standard stream of the archive to be checked.

`BitArchiveReader::isHeaderEncrypted` checks whether the archive is fully encrypted (header included), and hence it needs a password to be opened.

`BitArchiveReader::isEncrypted` checks whether the archive contains only encrypted files, but can be opened without a password (plain headers).

**Comment by rikyoz** [comment]
Implemented in [v4.0.0](https://github.com/rikyoz/bit7z/releases/tag/v4.0.0).
